### PR TITLE
New docs website / `Doc::CopyButton` component

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

I needed the `CopyButton` component in the Icons grid, so I implemented it and opened a PR for it, so that once approved @MelSumner @alex-ju can also use them in their components (`ColorSwatch` and `CodeBlock`).

For context: I've used https://github.com/jkusa/ember-cli-clipboard because it's widely used in our codebase and well maintained.

### :hammer_and_wrench: Detailed description

In this PR I have:
- added the `Doc::CopyButton` component
- added a testing page for `Doc::CopyButton` component

👉 👉 👉 **Preview**:  https://hds-website-git-new-docs-website-copy-button-hashicorp.vercel.app/testing/components/copy-button/

### :camera_flash: Screenshots

**In code**
screenshot_2153

**In design (Figma file)**
screenshot_2154

### :link: External links

JIRA ticket: https://hashicorp.atlassian.net/browse/HDS-1212 
Figma file: https://www.figma.com/file/Ky0qWjvHZR3je1lCBlzNB1/HDS-website?node-id=1455%3A95568&t=YNJcoWCVScYCh1p5-1

### 👀 How to review

👉 Review commit-by-commit or by files changed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: website/app/components/doc/copy-button/index.hbs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<CopyButton
+  class={{this.classNames}}
+  @text={{this.textToCopy}}
+  @onSuccess={{this.onSuccess}}
+  @onError={{this.onError}}
+>
+  {{#if this.textToShow}}
+    <span class="doc-copy-button__visible-value">{{this.textToShow}}</span>
```
Comment (alex-ju) [inline website/app/components/doc/copy-button/index.hbs:8]: I imagine we call this `visible-value` to distinguish between what's copied and what's shown

**Review by alex-ju — APPROVED** [review]
Looking good! Thanks for taking the time to work on this!
I noticed the Metro font from our sans stack has a lower baseline and cap/x-height than most fonts, which causes a visual misalignment with icons which may bite us in scenarios like this one – but that's an issue for another day

File: website/app/components/doc/copy-button/index.hbs (hunk 1 above)
Comment (didoo) [reply website/app/components/doc/copy-button/index.hbs:8]: Yep! I imagine cases (eg a long text to copy / a long code snippet) in which we want to differentiate what the user sees and what they actually copy.

**Comment by didoo** [comment]
> I noticed the Metro font from our sans stack has a lower baseline and cap/x-height than most fonts, which causes a visual misalignment with icons which may bite us in scenarios like this one – but that's an issue for another day

@alex-ju yes this is a know issue (@heatherlarsen mentioned it different times, and is known also on the marketing side)

I think once everything is in place and all the designs are final, we can do a session all together and have a pass across the entire website and see where the text visual alignment need some small tweaks (I am position relative with negative top, or negative margin top, depending on the cases).

**Review by MelSumner — APPROVED** [review]
Awesome, this is really useful. 👍

**Comment by didoo** [comment]
I am going to merge this, so @MelSumner and @alex-ju can start to use it in their PRs. If there are changes/tweaks to do, we can do them in follow-up PRs.

**Pull request merged**